public/app/plugins/datasource/grafana-azure-monitor-datasource/utils/migrateQuery.ts

```diff
@@ -146,7 +146,10 @@ function migrateDimensionToResourceObj(query: AzureMonitorQuery): AzureMonitorQu
         resourceGroup: details?.resourceGroup,
         metricNamespace: details?.metricNamespace,
         resourceName: details?.resourceName,
-        resourceUri: undefined,
+        resourceUri:
+          details?.subscription && details?.resourceGroup && details?.resourceName
+            ? undefined
+            : query.azureMonitor.resourceUri,
```

Comment:
if we leave the `resourceUri` , does it mean the query will be migrated again next time? As `query.azureMonitor?.resourceUri` will be true?
Maybe I'm misunderstanding what you're trying to achieve here 😅

Reply:
If the `resourceUri` is a template variable then we'd require the query to be migrated every time to account for any changes to the variable 😊 

The goal of this PR is to ensure that if a template variable is used for the `resourceUri` (as is the case for the support escalation) then it should be appropriately migrated to the relevant components.

Reply:
well, we _don't_ need to migrate it in the case it's a template variable, since it will never be parsed. Maybe you can update the condition to something like:

```
  if (query.azureMonitor?.resourceUri && !query.azureMonitor.resourceUri.startsWith('$')) {
```

To be clear that template variables won't be migrated (but the result will be the same than with this code).

What would be nice, is to modify the "Advance" section of the resource picker to show the resource URI (in this case, the template variable) rather than the empty fields that they are going to see for the Subscription + Resource Group + Resource Name. But you can do that in a separate PR

Reply:
Gotcha! Thanks, I think Andres' suggestion makes sense, maybe we can add some comment in the code for the next person that is trying to make sense of it?

```
// migrating if the query is using the deprecated `resourceURI` prop and it's not a template variable
```

Could we always set the uri as `undefined` if we change the condition then? 
If not, could we maybe extract `details?.subscription && details?.resourceGroup && details?.resourceName` and give it a name that would help us understand what we are trying to achieve, such as `isWellFormedURI` or `isNotTemplateVariableURI`?

Reply:
I think it's helpful to not set always the uri as undefined, to cover corner cases, like template variables covering weird parts of the resource URI (e.g. `/subscriptions/$restOfTheThing`). +1 to define the condition as a variable.

Reply:
I've made the suggested changes and will set this to auto-merge 😊 Thanks!